buildenv/jenkins/omrbuild.groovy

```diff
@@ -20,26 +20,38 @@
  * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
  *******************************************************************************/
 
-def setBuildStatus(String message, String state, String sha) {
-    context = "continuous-integration/eclipse-omr/branch/${params.BUILDSPEC}"
-    step([
-        $class: "GitHubCommitStatusSetter",
-        contextSource: [$class: "ManuallyEnteredCommitContextSource", context: context],
-        errorHandlers: [[$class: "ChangingBuildStatusErrorHandler", result: "UNSTABLE"]],
-        commitShaSource: [$class: "ManuallyEnteredShaSource", sha: sha ],
-        statusBackrefSource: [$class: "ManuallyEnteredBackrefSource", backref: "${BUILD_URL}flowGraphTable/"],
-        statusResultSource: [$class: "ConditionalStatusResultSource", results: [[$class: "AnyBuildResult", message: message, state: state]] ]
-    ]);
-}
+defaultCompile = 'make -j4'
+defaultReference = '${HOME}/gitcache'
+autoconfBuildDir = '.'
+cmakeBuildDir = 'build'
+workspaceName = 'Build'
+scmVars = null
+
+dockerImage = null
+dockerImageName = (params.IMAGE_NAME) ? params.IMAGE_NAME : "buildomr"
+
+os = (params.OS) ? params.OS : "ubuntu20"
+arch = (params.ARCH) ? params.ARCH : "x86_64"
 
-def defaultCompile = 'make -j4'
-def defaultReference = '${HOME}/gitcache'
-def autoconfBuildDir = '.'
-def cmakeBuildDir = 'build'
-def workspaceName = 'Build'
+buildSpec = (params.BUILDSPEC) ? params.BUILDSPEC : null
+cmdLine = (params.ghprbCommentBody) ? params.ghprbCommentBody : null
+pullId = (params.ghprbPullId) ? params.ghprbPullId : null
 
-def SPECS = [
+cgroupV1Specs = ["linux_x86"]
+cgroupV2Specs = ["linux_x86-64"]
+dockerSpecs = ["linux_x86-64"]
```

Comment:
Are there advantages to having these vars here rather than including them in `SPECS`?

Reply:
If it goes in `SPECS`, then we will need three new fields in each specification: `cgroupv1Default`, `cgroupv2Default` and `dockerDefault`. Also, `cgroup.v[1|2]` and `docker` labels will only apply to Linux specifications. The current approach is better in terms of space, and it is easier to manage the default specs that need to be run in either modes. Although an assertion has been added, we can easily determine conflicts i.e. one spec should not have both cgroup.v1 and cgroup.v2 labels.